src/SkyDrop.iOS/Common/iOSUtil.cs

```diff
@@ -17,5 +22,44 @@ public static Task<UIImage> BitMatrixToImage(BitMatrix bitMatrix)
                 return renderer.Render(bitMatrix, ZXing.BarcodeFormat.QR_CODE, "");
             });
         }
+
+        public static void RegisterForLocalNotifications()
+        {
+            var settings = UIUserNotificationSettings.GetSettingsForTypes(UIUserNotificationType.Alert, null);
+            UIApplication.SharedApplication.RegisterUserNotificationSettings(settings);
+        }
+
+        public static void ShowUploadFinishedNotification(FileUploadResult uploadResult, string filename)
+        {
+            switch (uploadResult)
+            {
+                case FileUploadResult.Success:
+                    ShowNotification("File published successfully (tap to view)", filename);
+                    break;
+                case FileUploadResult.Fail:
+                    ShowNotification("Upload failed", filename);
+                    break;
+                case FileUploadResult.Cancelled:
+                    break;
+            }
+        }
+
+        public static void ShowNotification(string title, string message)
+        {
+            if (uploadNotification != null)
+                UIApplication.SharedApplication.CancelLocalNotification(uploadNotification);
+
+            uploadNotification = new UILocalNotification
+            {
+                FireDate = NSDate.FromTimeIntervalSinceNow(2),
+                AlertTitle = title,
+                AlertBody = message,
+                ApplicationIconBadgeNumber = 0,
+                SoundName = UILocalNotification.DefaultSoundName,
+                UserInfo = new NSDictionary()
+            };
+
+            UIApplication.SharedApplication.ScheduleLocalNotification(uploadNotification);
+        }
     }
```

Comment:
Wondering if the notification should show when the app is in foreground?

Reply:
I think it would be complicated to implement